src/vm/frames.cpp

```diff
@@ -1001,6 +1001,31 @@ VOID GCFrame::Pop()
 #endif
 }
 
+#if !defined(FEATURE_PAL) && !defined(CROSSGEN_COMPILE)
+// GCFrame destructor removes the GCFrame from the current thread's explicit frame list.
+// This prevents issues in functions that have HELPER_METHOD_FRAME_BEGIN / END around 
+// GCPROTECT_BEGIN / END and where the C++ compiler places some local variables over 
+// the stack location of the GCFrame local variable after the variable goes out of scope. 
+GCFrame::~GCFrame()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    if (m_Next != NULL)
```

Comment:
Is this check for null needed? GCFrame is always pushed in constructor, so we should never come here with m_Next == NULL

Reply:
Yes, it is. When runtime calls WhateverFrame::GetMethodFrameVPtr, it creates a dummy instance of the frame, then gets the vtable and then destroys it. This check ensures that we don't try to pop such frames.

Reply:
ok